cobalt/user_agent_string.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cobalt/user_agent_string.h"
+
+#include "base/strings/stringprintf.h"
+
+namespace cobalt {
+
+namespace {
+
+const char kUnknownFieldName[] = "Unknown";
+
+}  // namespace
+
+UserAgentPlatformInfo GetUserAgentPlatformInfoFromSystem() {
+  UserAgentPlatformInfo platform_info;
+  return platform_info;
```

Comment:
Could we add a `UserAgentPlatformInfo::Create()` and nuke
this `GetUserAgentPlatformInfoFromSystem()` ?